### ISSUE TITLE
S3 file upload throws exception 

### ISSUE THREAD

**Issue by florinppgmail** [thephpleague/flysystem#902]
First of all I would like to say "thank you" for this wonderful tool. I'm submitting this issue for maybe a further improvement.

I'm uploading files to S3 bucket, using FlySystem within Laravel 5.6.
All good on local dev environment, Homestead. On production there is a problem with the upload. 
The error shown is:

```
exception : "ErrorException"
file : "/full/path/to/laravel/app/vendor/laravel/framework/src/Illuminate/Filesystem/FilesystemAdapter.php"
"fopen(/full/path/to/laravel/app/public): failed to open stream: Is a directory"
```

By conducting some tests I can confirm that the problem was with the file size of the uploaded file.

Is there a possibility to handle the issue by throwing a more explicit Exception message.

**Comment by frankdejonge** [comment]
This looks like you're trying to upload to a directory, you need to upload to a file location.

**Comment by florinppgmail** [comment]
That's my point: "looks like ... ". The error message is pointing in a wrong direction.

I didn't had time to investigate thoroughly the library code to see exactly why is behaving like this when the file size exceeds. 
But when I increased the max allowed upload size and restarted the web server (in this case was apache and I increased the size from 2Mb to 100Mb), the message went away and the upload went to the S3 bucket successfully.
